library/plugin/reader.h

```diff
@@ -132,6 +133,38 @@ class reader
   virtual void applyCustomImporter(vtkImporter*, const std::string&) const
   {
   }
+
+  /**
+   * Set a reader option
+   * Return true if the option was found (and set), false otherwise
+   */
+  bool setReaderOption(const std::string& name, const std::string& value)
+  {
+    auto iter = this->ReaderOptions.find(name);
+    if (iter == this->ReaderOptions.end())
+    {
+      return false;
+    }
+    iter->second = value;
+    return true;
+  }
+
+  /**
+   * Return the list of all reader option names
+   */
+  std::vector<std::string> getAllReaderOptionNames()
+  {
+    std::vector<std::string> keys;
+    keys.reserve(this->ReaderOptions.size());
+    for (const auto& [key, value] : this->ReaderOptions)
+    {
+      keys.push_back(key);
+    }
+    return keys;
+  }
+
+protected:
+  std::map<std::string, std::string> ReaderOptions;
```

Comment:
why not unordered_map?

Reply:
no reason, tbf considering the size of these maps, i dont think it matters